### ISSUE TITLE
NameSpace not found

### ISSUE THREAD

**Issue by avinashdaiict** [i18next/i18next-scanner#31]
Could someone please tell why I am getting this error when I try to pass the argument "App key : "

This is the error

Please ensure the namespace "App Key" exists in the ns.namespaces[] of your i18next-scanner options: {"ns":"App Key","key":"","defaultValue":"App Key:"}

**Comment by cheton** [comment]
It looks like you're using the [Key Based Fallback](http://i18next.com/translate/keyBasedFallback/) style for translations (e.g. gettext). If yes, you have to set both `nsSeparator` and `keySeparator` to false in i18next-scanner as well as your i18next options. For example:
```js
var scanner = require('i18next-scanner');
var vfs = require('vinyl-fs');
var options = {
    nsSeparator: false,
    keySeparator: false,
        :   :   :
    // See options at https://github.com/i18next/i18next-scanner#options
};
vfs.src(['/path/to/src'])
    .pipe(scanner(options))
    .pipe(vfs.dest('/path/to/dest'));
```

See also:
- https://github.com/i18next/i18next-scanner#nsseparator
- https://github.com/i18next/i18next-scanner#keyseparator

**Comment by cheton** [comment]
I can refine the error message to make it more clear. For example:
```
Make sure you have added the namespace "{{ns}}" to options.ns array. Otherwise, set 
options.nsSeparator and/or options.keySeparator to false if you're using key based fallback for
translations (e.g. gettext).
```

**Comment by avinashdaiict** [comment]
Yeah I got the issue ....But the problem is that when  I have some text which have an colon ..say 
"Student Id:201201153"  then the parser thinks that it is key value pair and breaks it ....
Now if I pass false as the nsseparator the parser runs fine except that it stops checking for duplicate entries which is also not a desired use case...

Is there any way that I can bypass '':" in a strings 

thanks for your help

**Comment by cheton** [comment]
What do you mean by <i>`"the parser runs fine except that it stops checking for duplicate entries"`</i>? 

Could you provide me a practical example including scanning contents and your i18next options, so that I will  know what you've encountered. Thanks.

**Comment by avinashdaiict** [comment]
Hey @cheton  
Lets take an example case 

"App Key: 201201153": "App key :201201153" in "three " of the files 
with the same values in both the key and value pair 

Now if I pass nsselector as the default (:)  then at the time of parsing there is an error shown that shows
 
**Please ensure the namespace "App Key" exists in the ns.namespaces[] of your i18next-scanner options: {"ns":"App Key","key":"","defaultValue":"App Key:"}**

This may be because the parser splits at ":" and so it thinks "App Key " is an existing key but it is not the case so the error

Now if I pass nsseparator as false what happens is that the parsing goes fine but as I am having the same label (App key:) in three places there are repetition in the output because the nsseperator is null .

So how do I get past this use case ?

**Comment by cheton** [comment]
Hi @avinashdaiict,

What's your version of i18next-scanner?
I just realized that your message string "<b>Please ensure the namespace ...</b>" is only available in earlier v0.6.1, which is an outdated version and it didn't provide support for key-based fallback.

Could you upgrade your i18next-scanner to v1.3.1, and checkout migration guide if you're upgrading from earlier versions.
https://github.com/i18next/i18next-scanner/wiki/Migration-Guide 

Note that  `nsseparator` and `keyseparator` have been renamed to `nsSeparator` and `keySeparator` since v1.0.

**Comment by cheton** [comment]
May I know which version of i18next library were currently using in your project? The key-based fallback is available since i18next v2.1.0.

See changelog - https://github.com/i18next/i18next/blob/master/CHANGELOG.md#210
